### DISCUSSION TITLE
METForcing data datewise or variablewise ? (NGFS same as GDAS metforcing)

### DISCUSSION THREAD

**Discussion by hashmifatima** [NASA-LIS/LISF#1304] [General]
Hi 

I have a reanalysis metforcing data NGFS (25 km, 6 hourly) which is being generated in real time using Global Data Assimilation Forecasting System (GDAFS). Forcing data files are available in the format :
ngfs_reanl_HR_APCP-sfc_2002100100-2002103118.grb2
ngfs_reanl_HR_PRES-sfc_2002100100-2002103118.grb2
ngfs_reanl_HR_SPFH-2m-agl_2002100100-2002103118.grb2
ngfs_reanl_HR_TMP-2m-agl_2002100100-2002103118.grb2
ngfs_reanl_HR_UGRD-10m-agl_2002100100-2002103118.grb2
ngfs_reanl_HR_VGRD-10m-agl_2002100100-2002103118.grb2

This NGFS data is having same resolutionas GDAS data, GDAS data available file format is this:
 2002103112.gdas1.sfluxgrbf03.sg
2002103112.gdas1.sfluxgrbf00.sg
2002103106.gdas1.sfluxgrbf09.sg
 2002103106.gdas1.sfluxgrbf06.sg
2002103106.gdas1.sfluxgrbf03.sg
2002103106.gdas1.sfluxgrbf00.sg
 2002103100.gdas1.sfluxgrbf09.sg
2002103100.gdas1.sfluxgrbf06.sg
 2002103100.gdas1.sfluxgrbf03.sg
2002103100.gdas1.sfluxgrbf00.sg
2002103018.gdas1.sfluxgrbf09.sg
2002103018.gdas1.sfluxgrbf06.sg
 2002103018.gdas1.sfluxgrbf03.sg
2002103118.gdas1.sfluxgrbf09.sg
2002103118.gdas1.sfluxgrbf06.sg
2002103118.gdas1.sfluxgrbf03.sg
2002103118.gdas1.sfluxgrbf00.sg
2002103112.gdas1.sfluxgrbf09.sg
2002103112.gdas1.sfluxgrbf06.sg

GDAS data file contains all input variables in a single file separated by date/time but NGFS data file is like each variable is having separate file for total time period. will this format of NGFS data is accepted by GDAS metforcing criteria as both are generated from the same source?
Does LIS accept variables wise input files or it takes datewise files as well?

Thanks, 
Hashmi

**Comment by dmocko** [comment]
Hi @hashmifatima 

Yes, LIS can handle metforcing files separated by variable as well as files with multiple dates within.

You would just need to write the metforcing plugin for this dataset such that it opens each file when reading the variable - and that it reads the correct time slice when LIS is at that time.

Depending on how you write your metforcing reader, there may be a slowdown of the code as it finds the correct time within such a large file.  However, it should still work if you are careful when writing the plugin.

Good luck!

**Reply by hashmifatima** [reply]
Hi @dmocko 

Thanks for the reply. Can I use "genmetforc" for NGFS forcing"? 

There is a code for metforcing plugin in this directory: ~/LISF-7.5.3-557WW/lis/plugins/LIS_metforcing_pluginMod.F90
Individual metforcing reader codes are available in seperate directories. ~/LISF-7.5.3-557WW/lis/metforcing 

If I am writing a new reader then I need to create at both the places?

Regards
Hashmi

**Reply by dmocko** [reply]
Hi @hashmifatima 

"genmetforc" is for forcing datasets that were created using LDT.  Since you already have the forcing files, you don't need to create them through LDT.

Instead, you need to write a new metforcing reader.  You can use an existing metforcing reader as a guide/template.

You would create a new directory in LISF/lis/metforcing for the reader, put the code in there, and then edit the LISF/lis/metforcing/default.cfg file to put in the directory path to the new code.

You also need to update the LISF/lis/plugins/LIS_metforcing_pluginMod.F90 file (as you note) for the new forcing, as well as add it into LISF/lis/plugins/LIS_pluginIndices.F90.

Again, you can just follow/copy an existing metforcing reader as a guide when adding in this new forcing.
